scripts/azureml-assets/azureml/assets/update_assets.py

```diff
@@ -63,6 +65,40 @@ def release_tag_exists(asset_config: assets.AssetConfig, release_directory_root:
     return asset_config.full_name in repo.tags
 
 
+
+def validate_new_release(asset_config: assets.AssetConfig, release_directory_root: Path) -> bool:
+    """Check repo to see if an asset's previous version was released if a latest tag exists.
+
+    Args:
+        asset_config (assets.AssetConfig): Asset config
+        release_directory_root (Path): Release branch location
+
+    Returns:
+        bool: True if the previous tag exists, False otherwise
+    """
+    if asset_config.type != AssetType.ENVIRONMENT:
+        return True
+
+    repo = Repo(release_directory_root)
+    commits_tags = {}
+
+    for tag in repo.tags:
+        if tag.name.startswith(f"{asset_config.partial_name}/"):
+            commits_tags[tag.commit] = tag
+
+    ordered_commits = sorted(commits_tags.keys(), key=lambda c: c.authored_datetime)
+
+    # If no tags exist for this asset, return True
+    if len(ordered_commits) == 0:
+        return True
+
+    # Get the latest tag
+    latest_tag = commits_tags[ordered_commits[-1]].name
+    latest_version = int(latest_tag.split("/")[2])
+
+    return latest_version + 1 == int(asset_config.version)

```

Comment:
What do you think about this type of approach? I just figured out that we're grabbing commit timestamps this way, but not tag timestamps, which will follow by some amount of time.

```suggestion
    tags = [t for t in repo.tags if tag.name.startswith(f"{asset_config.partial_name}/")]
    if not tags:
       # No releases
        return True

    # Get the latest tag
    ordered_tags = sorted(tags, key=lambda t: t.tag.tagged_date)
    latest_tag = ordered_tags[-1].name
    _, _, latest_version = assets.AssetConfig.parse_full_name(latest_tag)

    return int(latest_version) + 1 == int(asset_config.version)
```

Note that the last line just compares the asset version against the latest version. Perhaps this function should just return the last version (that should be released) that you can then check against MCR?

Reply:
Does checking against MCR work similarly to _get_latest_image_suffix?

Reply:
I think it will be more like [_get_latest_tag_or_digest](https://github.com/Azure/azureml-assets/blob/702a9a7d96c559bc8854179b7aadff7f49a57e6c/scripts/azureml-assets/azureml/assets/environment/pin_image_versions.py#L57-L62), in that we want to make a manifest call using a specific tag, to see whether the tag exists.